interpreter/cling/include/cling/Interpreter/Interpreter.h

```diff
@@ -281,7 +281,22 @@ namespace cling {
     ///\param[in] argv - arguments passed when driver is invoked.
     ///\param[in] llvmdir - ???
     ///\param[in] noRuntime - flag to control the presence of runtime universe
-    Interpreter(int argc, const char* const *argv, const char* llvmdir = 0, bool noRuntime = false);
+    ///\param[in] isChildInterp - flag to control if this is a child interpreter or not.
+    ///
+    Interpreter(int argc, const char* const *argv,
+                const char* llvmdir = 0, bool noRuntime = false, bool isChildInterpreter = false);
```

Comment:
Can we have the union of the two interfaces in a non-public (likely private) function, instead of adding a parameter to a public interface that doesn't make sense when called by users? Or do you think there's a use case for having this constructor called with `isChildInterpreter = true` in user code? If not let's move the common functionality out, and keep the public constructors with as few parameters as possible. (There are already too many...)